openapi_tester/exceptions.py

```diff
@@ -8,33 +8,40 @@ class DocumentationError(AssertionError):
     Custom exception raised when package tests fail.
     """
 
-    def __init__(
+    def __init__(  # pylint: disable=too-many-arguments
         self,
         message: str,
         response: Any,
         schema: dict,
         hint: str = "",
         reference: str = "",
+        show_expected: bool = True,
     ) -> None:
         from openapi_tester.schema_converter import SchemaToPythonConverter
 
-        converted_schema = SchemaToPythonConverter(schema, with_faker=False).result
-        super().__init__(
-            self.format(
-                response=self._sort_data(response),
-                example_item=self._sort_data(converted_schema),
-                hint=hint,
-                message=message,
-                reference=reference,
+        if show_expected:
+            converted_schema = SchemaToPythonConverter(schema, with_faker=False).result
+            super().__init__(
+                self.format(
+                    response=self._sort_data(response),
+                    example_item=self._sort_data(converted_schema),
+                    hint=hint,
+                    message=message,
+                    reference=reference,
+                )
             )
-        )
+        else:
+            super().__init__(message)
 
     @staticmethod
     def _sort_data(data_object: Any) -> Any:
         if isinstance(data_object, dict):
             return dict(sorted(data_object.items()))
         if isinstance(data_object, list):
-            return sorted(data_object)
+            try:
+                return sorted(data_object)
```

Comment:
this is redundant - we already check for isinstance above, that why i removed the try catch block before.

Reply:
The removal broke the logic (and a test). Sorting can break with `TypeError` depending on the contents of the list.

Reply:
```python
sorted([1,2,4,3])
Out[2]: [1, 2, 3, 4]
sorted([{},{}])
Traceback (most recent call last):
  File "/Users/sondrelg/Documents/django-openapi-tester/.venv/lib/python3.9/site-packages/IPython/core/interactiveshell.py", line 3343, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-3-96c4907b4da2>", line 1, in <module>
    sorted([{},{}])
TypeError: '<' not supported between instances of 'dict' and 'dict'
```

Reply:
In this case we want to default to returning them unsorted, which is what we achieve with the try/except block

Reply:
Cool